Framework/DataHandling/src/SaveNXcanSASBase.cpp

```diff
@@ -84,87 +96,192 @@ void SaveNXcanSASBase::initStandardProperties() {
      otherwise check that the input is a valid 2D workspace.*/
-  Mantid::API::Workspace_sptr const workspace = getProperty("InputWorkspace");
+  const Workspace_sptr workspace = getProperty(StandardProperties::INPUT_WORKSPACE);
   auto valid2DWorkspace = [](const Workspace_sptr &ws) {
     return ws && std::dynamic_pointer_cast<const Mantid::DataObjects::Workspace2D>(ws);
   };
   if (workspace->isGroup()) {
-    auto const &groupItems = std::dynamic_pointer_cast<WorkspaceGroup>(workspace)->getAllItems();
+    const auto &groupItems = std::dynamic_pointer_cast<WorkspaceGroup>(workspace)->getAllItems();
     if (std::any_of(groupItems.cbegin(), groupItems.cend(),
-                    [&](auto const &childWs) { return !valid2DWorkspace(childWs); })) {
-      result.emplace("InputWorkspace", "All input workspaces in the input group must be a Workspace2D.");
+                    [&](const auto &childWs) { return !valid2DWorkspace(childWs); })) {
+      result.emplace("InputWorkspace",
+                     "All input workspaces in the input group must be a Workspace2D wit numeric axis.");
     }
   } else {
     if (!valid2DWorkspace(workspace)) {
-      result.emplace("InputWorkspace", "The InputWorkspace must be a Workspace2D.");
+      result.emplace("InputWorkspace", "The InputWorkspace must be a Workspace2D with numeric axis.");
     }
   }
 
   // Transmission data should be 1D
-  Mantid::API::MatrixWorkspace_sptr transmission = getProperty("Transmission");
-  Mantid::API::MatrixWorkspace_sptr transmissionCan = getProperty("TransmissionCan");
+  MatrixWorkspace_sptr transmission = getProperty(StandardProperties::TRANSMISSION);
+  MatrixWorkspace_sptr transmissionCan = getProperty(StandardProperties::TRANSMISSION_CAN);
 
-  auto checkTransmission = [&result](const Mantid::API::MatrixWorkspace_sptr &trans, const std::string &propertyName) {
+  auto checkTransmission = [&result](const MatrixWorkspace_sptr &trans, const std::string &propertyName) {
     if (trans->getNumberHistograms() != 1) {
       result.emplace(propertyName, "The input workspaces for transmissions have to be 1D.");
     }
   };
 
   if (transmission) {
-    checkTransmission(transmission, "TransmissionCan");
+    checkTransmission(transmission, StandardProperties::TRANSMISSION);
   }
   if (transmissionCan) {
-    checkTransmission(transmissionCan, "TransmissionCan");
+    checkTransmission(transmissionCan, StandardProperties::TRANSMISSION_CAN);
+  }
+
+  return result;
+}
+
+std::map<std::string, std::string>
+SaveNXcanSASBase::validatePolarizedInputWorkspace(const std::vector<std::string> &spinVec) const {
+  std::map<std::string, std::string> result;
+  const Workspace_sptr workspace = getProperty(StandardProperties::INPUT_WORKSPACE);
+
+  if (!workspace->isGroup())
+    result.emplace(StandardProperties::INPUT_WORKSPACE,
+                   "Input Workspaces for polarized data can only be workspace groups.");
+  else {
```

Comment:
Rather than a big if/else, it may be worth simply returning here instantly as we know we can't do any more checks. 
```suggestion
  if (!workspace->isGroup()) {
    result.emplace(StandardProperties::INPUT_WORKSPACE,
                   "Input Workspaces for polarized data can only be workspace groups.");
    return result;
  }

```